File: flipEquivalent.py
class Solution:
    def flipEquiv(self, root1: TreeNode, root2: TreeNode) -> bool:
        if root1 == root2:
            return True
        if root1 is None or root2 is None or root1.val != root2.val:
            return False
        return (self.flipEquiv(root1.left, root2.left) and self.flipEquiv(root1.right, root2.right) or
                self.flipEquiv(root1.left, root2.right) and self.flipEquiv(root1.right, root2.left))

File: flipEquivalent.cpp
struct TreeNode{
   int val;
   TreeNode* left;
   TreeNode* right;
};

class Solution {
public:
     bool flipEquivalent(TreeNode* root1, TreeNode *root2){
          if(root1 == root2)
               return true;

          if(root1 == null || root2 == null || root1.val != root2.val)
              return false;

          return (flipEquivalent(root1.left, root2.left) and flipEquivalent(root1.right, root2.right)) ||
           (flipEquivalent(root1.left, root2.right) and flipEquivalent(root1.right, root2.left));
     }
};
